### PR TITLE
Pools, Agents  and Queues retrieval commands

### PR DESCRIPTION
**Pull request by atbagga**
Please make sure the code is following contribution guidelines in CONTRIBUTING.md

 - [ ] : This PR has a corresponding issue open in the Repository. #504
 - [ ] : Approach is signed off on the issue.

### CODE REVIEW

File: azure-devops/azext_devops/dev/pipelines/agent_pool_queue.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+def list_pools(pool_name=None, pool_type=None, action=None, properties=None, organization=None, detect=None):
+    """ (PREVIEW) List agent pools
+    :param pool_name: Filter the list with matching pool name.
+    :type pool_name: str
+    :param pool_type: Filter the list with type of pool.
+    :type pool_type: str
+    :param action: Filter the list with user action permitted.
+    :type action: str
+    :param properties: Filter the list by agent pool properties. Comma separated list.
```
Comment (ishitam8) [inline azure-devops/azext_devops/dev/pipelines/agent_pool_queue.py]: An example of properties list would be helpful here

File: azure-devops/azext_devops/test/pipelines/test_pipeline_agents.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+except ImportError:
+    # Attempt to load mock (works on Python version below 3.3)
+    from mock import patch
+
+from azext_devops.dev.common.services import clear_connection_cache
+from azext_devops.test.utils.authentication import AuthenticatedTests
+from azext_devops.devops_sdk.v5_1.task_agent.task_agent_client import TaskAgentClient
+from azext_devops.dev.pipelines.agent_pool_queue import list_pools, show_pool, list_agents, show_agent, list_queues, show_queue
+
+class TestPipelinesRunsArtifactsMethods(AuthenticatedTests):
```
Comment (ishitam8) [inline azure-devops/azext_devops/test/pipelines/test_pipeline_agents.py]: TestPipelinesAgentPoolQueue ?

File: azure-devops/azext_devops/dev/pipelines/agent_pool_queue.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    :type agent_name: str
+    :param include_capabilities: Whether to include the agents' capabilities in the response.
+    :type include_capabilities: bool
+    :param include_assigned_request: Whether to include details about the agents' current work.
+    :type include_assigned_request: bool
+    :param include_last_completed_request: Whether to include details about the agents' most recent completed work.
+    :type include_last_completed_request: bool
+    :param properties: Filter which custom properties will be returned. Comma separated list.
+    :type properties: str
+    :param demands: Filter by demands the agents can satisfy. Comma separated list.
```
Comment (ishitam8) [inline azure-devops/azext_devops/dev/pipelines/agent_pool_queue.py]: Example for 'demands' param as well

File: azure-devops/azext_devops/dev/pipelines/agent_pool_queue.py (hunk 1 above)
Comment (atbagga) [reply azure-devops/azext_devops/dev/pipelines/agent_pool_queue.py]: Will ask the team owning the API about how the properties look like.. I haven't seen an example yet where the properties are returned.

File: azure-devops/azext_devops/dev/pipelines/agent_pool_queue.py (hunk 3 above)
Comment (atbagga) [reply azure-devops/azext_devops/dev/pipelines/agent_pool_queue.py]: Will ask the team owning the API about how the filtering with demands work.

**Pull request merged**